### PR TITLE
Introduce class `ipl\Web\Compat\StyleWithNonce`

### PR DESCRIPTION
**Pull request by nilmerg**

### CODE REVIEW

File: src/Compat/StyleWithNonce.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * with Icinga Web &lt; 2.12 and with CSP support in &gt;= 2.12
+ */
+class StyleWithNonce extends Style
+{
+    public function getNonce(): ?string
+    {
+        if ($this->nonce === null) {
+            $this->nonce = version_compare(Version::VERSION, '2.12.0', '>=')
+                ? Csp::getStyleNonce()
+                : '';
```
Comment (yhabteab) [inline src/Compat/StyleWithNonce.php:20]: Why empty string instead of null?

**Review by sukhwinder33445 — DISMISSED** [review]

File: src/Compat/StyleWithNonce.php (hunk 1 above)
Comment (nilmerg) [reply src/Compat/StyleWithNonce.php:20]: Because otherwise the version compare and method call would needlessly occur in case of < 2.12

Comment (nilmerg) [reply src/Compat/StyleWithNonce.php:20]: ..at every getter call

**Review by yhabteab — APPROVED** [review]

File: src/Compat/StyleWithNonce.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * Use this class to define inline style which is compatible
+ * with Icinga Web &lt; 2.12 and with CSP support in &gt;= 2.12
+ */
+class StyleWithNonce extends Style
+{
+    public function getNonce(): ?string
+    {
+        if ($this->nonce === null) {
+            $this->nonce = version_compare(Version::VERSION, '2.12.0', '>=')
+                ? Csp::getStyleNonce()
```
Comment (nilmerg) [inline src/Compat/StyleWithNonce.php:19]: oh well, should also return `?? ''` i guess

**Pull request merged**